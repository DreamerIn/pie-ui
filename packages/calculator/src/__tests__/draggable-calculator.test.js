import React from 'react';
import DraggableCalculator from '../draggable-calculator';

describe('Renders a calculator component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<DraggableCalculator mode='basic' />)
  })

  it('Creates snapshot using enzyme', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('updates state.deltaPosition', () => {
    wrapper.instance().handleDrag(null, { deltaX: 10, deltaY: 10 });
    const deltaPosition = wrapper.state('deltaPosition');
    expect(deltaPosition).toEqual({ x: 10, y: 10 });
  })

});
