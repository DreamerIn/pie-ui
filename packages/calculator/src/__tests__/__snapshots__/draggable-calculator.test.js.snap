// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a calculator component Creates snapshot using enzyme 1`] = `""`;
